scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -574,7 +574,13 @@ class ScioContext private[scio] (val options: PipelineOptions, private var artif
     idAttribute: String,
     timestampAttribute: String): SCollection[(T, Map[String, String])] = {
     val io = PubsubIO.withAttributes[T](name, idAttribute, timestampAttribute)
-    this.read(io)(PubsubIO.ReadParam(isSubscription))
+    val read = this.read(io)(PubsubIO.ReadParam(isSubscription))
+
+    if (this.isTest && timestampAttribute != null) {
```

Comment:
@ClaireMcGinty i was wondering if we should to this inside `PubsubIOWithAttributes` because now since we have multiple ways of reading the behavior is not consistent, i.e:

```
sc.read(PubsubIO.withAttributes(name, idAttribute, timestampAttribute))(ReadParam(false)))
```

Reply:
talked IRL with @ClaireMcGinty and this will require some more thinking.

Reply:
will merge this, then refactor our test IO setup in a different PR.